plugins/org.eclipse.xtend.core/src/org/eclipse/xtend/core/jvmmodel/DispatchMethodCompileStrategy.java

```diff
@@ -205,11 +209,14 @@ protected void generateActualDispatchCall(JvmOperation dispatchOperation, JvmOpe
 		a.append(")");
 	}
 
-	/**
-	 * TODO: we assume that names from the inferred JVM model are the same in the Java code. Does that assumption
-	 * hold?
-	 */
 	protected String getVarName(JvmIdentifiableElement ex, IAppendable appendable) {
-		return ex.getSimpleName();
+		final String name = ex.getSimpleName();
+		if (name == null) {
+			return "__unknown__";
+		} else if (keywords.isJavaKeyword(name)) {
+			return name+"_";
```

Comment:
sry i right busy now. can you have a look. i took it from
org.eclipse.xtext.xbase.compiler.JvmModelGenerator.makeJavaIdentifier(String)